### PR TITLE
Add exception for repeated call to add provider methods

### PR DESCRIPTION
**Pull request by mikary**
Fix for #2090, alternate solution to #2271 
@rowanmiller @ajcvickers 

@bricelam - Could you verify that the SQLite changes are reasonable?

### CODE REVIEW

**Comment by ajcvickers** [comment]
:shipit: Consider pulling the boiler-plate code into a common place exposed as an "internal" utility method in core, Since we are going to do it in all our providers, other providers may want to reuse our method. But I'm also okay leaving as if if you think its better.

File: test/EntityFramework.Sqlite.Tests/SqliteTestHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        public new static SqliteTestHelpers Instance { get; } = new SqliteTestHelpers();
+
+        public override EntityFrameworkServicesBuilder AddProviderServices(EntityFrameworkServicesBuilder builder)
+        {
+            return builder.AddSqlite();
+        }
+
+        protected override void UseProviderOptions(EntityOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlite(new SqlConnection("Database=DummyDatabase"));
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.Tests/SqliteTestHelpers.cs]: `Database=:memory:` would keep it more isolated

Comment (bricelam) [inline test/EntityFramework.Sqlite.Tests/SqliteTestHelpers.cs]: Not that it should ever be used...

**Comment by bricelam** [comment]
Sqlite stuff looks fine

**Comment by mikary** [comment]
Updated based on feedback

**Comment by mikary** [comment]
Closing to create new PR as the intent of this PR has changed.

**Pull request closed without merging**